substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -3574,15 +3578,21 @@ impl<T: Config> Pallet<T> {
 	fn do_apply_slash(
 		member_account: &T::AccountId,
 		reporter: Option<T::AccountId>,
+		enforce_min_slash: bool,
```

Comment:
nit but in order to prevent spreading `do_apply_slash(..., bool)` across the code which is not very readable, how about creating an enum that helps with readability? something along the lines of 

```rust
enum SlashConfig {
 EnforceMinimum,
 DoNotEnforceMinimuim
}
```

Not sure about the enum and variant names, but in general

```rust
// reads better
do_apply_slash(..., SlashConfig::EnforceMinimum);

// than this
do_apply_slash(..., true);
```